### PR TITLE
shader_decompiler: Implemented (Partialy) TXQ

### PR DESCRIPTION
**Pull request by FernandoS27**
Used by SMO.

### CODE REVIEW

**Comment by lioncash** [comment]
Commit message for the relevant commit should include the game name.

**Comment by Hexagon12** [comment]
TXQ is also used by Mario+Rabbids, BotW, ARMS, Minecraft Story Mode.

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 1]
```diff
@@ -1847,6 +1847,22 @@ class GLSLGenerator {
                 WriteTexsInstruction(instr, coord, texture);
                 break;
             }
+            case OpCode::Id::TXQ: {
+                const std::string sampler = GetSampler(instr.sampler);
```
Comment (slashiee) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: error C2660: 'OpenGL::GLShader::Decompiler::GLSLGenerator::GetSampler': function does not take 1 arguments

**Comment by FernandoS27** [comment]
Should know be fixed but we got a problem with the samplers since the non-2D texture refactor.

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 2]
```diff
@@ -1920,6 +1920,26 @@ class GLSLGenerator {
                 WriteTexsInstruction(instr, coord, texture);
                 break;
             }
+            case OpCode::Id::TXQ: {
+                // TODO: the new commits on the texture refactor, change the way samplers work.
+                // Sadly, not all texture instructions specify the type of texture their sampler
+                // uses. This most be fixed at a later instance.
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: must

**Pull request merged**